### ISSUE TITLE
Cannot Publish an app using Code Magic after latest update

### ISSUE THREAD

**Issue by minashawky** [FlutterFlow/flutterflow-issues#3241]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When I publish an app via code magic I'm getting the following error: CocoaPods could not find compatible versions for pod "Firebase/Messaging"

<img width="1161" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/6050173/c97f9b98-432b-419c-852e-e21cc34cbbc2">

This just started to appear after the latest update FlutterFlow v4.1.63+ released June 26, 2024
Flutter version is 3.22.2

### Expected Behavior

Expecting the publishing to be successful

### Steps to Reproduce

Publish the app via Code Magic

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

AN/

### Visual documentation

N/A

### Environment

```markdown
- FlutterFlow version: v4.1.63+ released June 26, 2024
Flutter version is 3.22.2
- Platform: Web
- Browser name and version: Google Chrome Version 126.0.6478.63 (Official Build) (arm64)
- Operating system and version affected: macOS Sonoma 14.5
```

**Comment by minashawky** [comment]
Update: Since I was deploying from Github, I did some digging and had to:
- Upgrade Flutter
- Run flutter Clean
- Delete the ios/Pods folder and ios/Podfile.lock
- Run flutter pub get
- And now its deploying properly

**Comment by Steven-Simpla** [comment]
Hey @minashawky! I had a similar issue when I started trying to deploy from GitHub. I finally made it work by downloading the whole project, deleting a duplicate file in the build phases and then running pod install and flutter pub get.
But then I’m stuck in having to do the whole process through Xcode every time I’ll deploy.

I understand this can only be done through Xcode locally. How were you able to do it and still deploy from GitHub?
